libs/packages/sam-formly/src/lib/formly/wrappers/readonly.wrapper.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, ViewChild, ViewContainerRef } from '@angular/core';
+import { FieldWrapper } from '@ngx-formly/core';
+import { SdsFormlyTypes } from '../models/formly-types';
+/**
+ * @param {string} [to.required] Makes the field required
+ */
+@Component({
+  template: `
+    <div *ngIf="field.type === sdsFormlyTypes.READONLY || to.readonlyMode; else passThrough" [ngClass]="to.readonlyClass">
+      <span *ngIf="to.readonlyTemplate; else defaultTemplate" [innerHTML]="to.readonlyTemplate(field)"></span>
```

Comment:
Aside from the default readonly components, users can choose to also pass in custom template options if they need to render a more complex preview. In that case, the custom template passed in will be used.

Reply:
👍